### PR TITLE
DCOS-21772: add CI/CD

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
- [X] [Green build without deployment](https://jenkins.mesosphere.com/service/jenkins/job/Frontend/job/UI-Kit%20Pipeline/view/change-requests/job/PR-5/2/)
- [X] [Green build with deployment](https://jenkins.mesosphere.com/service/jenkins/job/Frontend/job/dcos-labs/job/UI-Kit%20Pipeline/view/change-requests/job/PR-5/15/console)
- [ ] Get Jenkins to report the status here

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM node:slim
```
Comment (weblancaster) [inline Dockerfile]: Let's use a specific version of node e.g `FROM node:8.9.4-alpine`

Comment (DanielMSchmidt) [reply Dockerfile]: Can you explain why?

**Comment by DanielMSchmidt** [comment]
Deployment doesn't seem to work because the repo is not cloned. I am honestly unsure if it wouldn't be better making it a `postrelease` hook instead 🤔 

https://jenkins.mesosphere.com/service/jenkins/job/Frontend/job/UI-Kit%20Pipeline/view/change-requests/job/PR-5/5/

(working around is possible but not so nice)

File: Dockerfile (hunk 1 above)
Comment (weblancaster) [reply Dockerfile]: Sure, There is no way to know what version of node we are using when `FROM node:slim` 
Would be nice to always install the version required/suggested and easy to know from the dockerfile

**Comment by weblancaster** [comment]
@DanielMSchmidt we want to have the storybook build/GH pages updated on each PR merged.
The UI Kit page should have the most recent from master. Basically a "nightly" build.

Let's try to figure out a way to have it done. I have a couple ideas how we can handle that so I will set a meeting for us including @Poltergeist

**Comment by weblancaster** [comment]
BTW, this is really good, thanks for opening the PRs! 🎉 ❤️

File: Dockerfile (hunk 1 above)
Comment (DanielMSchmidt) [reply Dockerfile]: Well, I don't have such a such a strong position there, I just like to keep these things up-to-date :) But I'll change it and we make sure to update it on a regular basis

**Comment by DanielMSchmidt** [comment]
@weblancaster I think the only thing we still need is to store a GH token for this repo in our Jenkins secrets 👍

File: Dockerfile (hunk 1 above)
Comment (weblancaster) [reply Dockerfile]: 🙏

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      }
+
+      post {
+        always {
+          archiveArtifacts 'dist/*'
+        }
+      }
+    }
+
+    stage('Deploy'){
```
Comment (weblancaster) [inline Jenkinsfile]: We should only deploy when a PR is merged so probably a separated hook. Seems like here will deploy everytime there is a PR successful merged or not merged, right @DanielMSchmidt?

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      when {
```
Comment (weblancaster) [inline Jenkinsfile]: We should only deploy when a PR is merged so probably a separated hook. Seems like here will deploy every time there is a PR successful merged or not merged, right @DanielMSchmidt?

Comment (DanielMSchmidt) [reply Jenkinsfile]: Nope, the when clause with the expression ```env.BRANCH_NAME == "master"``` you highlighted here causes Jenkins to skip the step if it is run from a PR and not on master

Comment (DanielMSchmidt) [reply Jenkinsfile]: You can see it in action in the builds I triggered: 

[Here you can see the deploy is not done](https://jenkins.mesosphere.com/service/jenkins/blue/organizations/jenkins/frontend%2Fdcos-labs%2FUI-Kit%20Pipeline/detail/PR-5/2/pipeline/)
bildschirmfoto 2018-03-27 um 09 11 36



[Here you can see I commented the line out, so it's done](https://jenkins.mesosphere.com/service/jenkins/blue/organizations/jenkins/frontend%2Fdcos-labs%2FUI-Kit%20Pipeline/detail/PR-5/15/pipeline)
bildschirmfoto 2018-03-27 um 09 11 39

Comment (weblancaster) [reply Jenkinsfile]: Awesome, thanks for explaining

Comment (DanielMSchmidt) [reply Jenkinsfile]: No worries 👍

**Comment by DanielMSchmidt** [comment]
@weblancaster I don't know if we need hold merge. It is done from a code perspective, only permissions in Jenkins / GH need to be updated :)

**Pull request merged**